src/basic/in-addr-util.c

```diff
@@ -464,3 +464,45 @@ int in_addr_mask(int family, union in_addr_union *addr, unsigned char prefixlen)
 
         return -EAFNOSUPPORT;
 }
+
+int in_addr_prefix_from_string(const char *p, int family, union in_addr_union *prefix, uint8_t *prefixlen) {
+        union in_addr_union buffer;
+        const char *e, *l;
+        uint8_t k;
+        int f, r;
+
+        assert(p);
+
+        e = strchr(p, '/');
+        if (e)
+                l = strndupa(p, e - p);
+        else
+                l = p;
+
+        r = in_addr_from_string_auto(l, &f, &buffer);
+        if (r < 0)
+                return r;
+
+        if (f != family)
+                return -1;
```

Comment:
We have both `in_addr_from_string` (which takes a `family` parameter), and `in_addr_from_string_auto` which does not. In this case we know what family we need, so it's OK to take a `family` parameter. But instead of calling `in_addr_from_string_auto`, call `in_addr_from_string(family, ...)`. This will be more efficient and `in_addr_from_string` will provide the right error code if necessary.